src/openrct2/ride/Ride.cpp

```diff
@@ -3108,8 +3108,16 @@ vehicle_colour ride_get_vehicle_colour(Ride* ride, int32_t vehicleIndex)
 {
     vehicle_colour result;
 
-    // Prevent indexing array out of bounds
-    vehicleIndex = std::min(vehicleIndex, MAX_CARS_PER_TRAIN);
+    if (ride->colour_scheme_type == VEHICLE_COLOUR_SCHEME_PER_VEHICLE)
```

Comment:
This is wrong. It should be a special case for when VEHICLE_COLOUR_SCHEME_SAME not for PER_VEHICLE